src/lib/logical_query_plan/lqp_translator.cpp

```diff
@@ -273,12 +273,20 @@ std::shared_ptr<AbstractOperator> LQPTranslator::_translate_alias_node(
   const auto input_node = alias_node->left_input();
   const auto input_operator = translate_node(input_node);
 
+  const auto& output_expressions = alias_node->output_expressions();
+  const auto output_expression_count = output_expressions.size();
   auto column_ids = std::vector<ColumnID>();
-  column_ids.reserve(alias_node->output_expressions().size());
+  column_ids.resize(output_expression_count);
 
-  for (const auto& expression : alias_node->output_expressions()) {
-    column_ids.emplace_back(input_node->get_column_id(*expression));
-  }
+  input_node->iterate_output_expressions([&](const auto column_id, const auto& input_expression) {
+    for (auto expression_id = ColumnID{0}; expression_id < output_expression_count; ++expression_id) {
+      const auto& expression = output_expressions[expression_id];
+      if (*expression == *input_expression) {
+        column_ids[expression_id] = column_id;
```

Comment:
I used to think the same, but we have [test cases with multiple aliases for the same column](https://github.com/hyrise/hyrise/blob/687d17316d0db77047a6f68e1a952198f74fef4e/resources/test_data/sqlite_testrunner_queries.sql#L129) that segfault otherwise.